### PR TITLE
Implement general infrastructure and Eve::CharacterInfo

### PR DESCRIPTION
**Pull request by amclain**
This PR implements the general infrastructure for the project, as well as the `Eve::CharacterInfo` endpoint in order to be able to test the infrastructure with a real endpoint. There are still some parts that are rough around the edges, as well as some code duplication that can be refactored out in later PR's. The idea of this PR is to get enough of the infrastructure's interface stable enough so that parallel development on the other endpoints can begin while the infrastructure is being fine-tuned.

### CODE REVIEW

**Comment by amclain** [comment]
@tweatherford and @tslogan, this is ready for review.

**Comment by amclain** [comment]
That code coverage decrease was expected, and is addressed in issue #4 (alluded to in the PR description) that consolidates duplicate code between `BaseItem` and `Row`.

File: lib/greeve/base_item.rb [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    end
+
+    # Fetch data from the API HTTP endpoint.
+    def fetch
+      url = "#{Greeve::EVE_API_BASE_URL}/#{endpoint}.xml.aspx"
+
+      query_params = @query_params
+
+      if @api_key && @api_vcode
+        query_params.merge!({
```
Comment (amclain) [inline lib/greeve/base_item.rb]: The query params need to be duped so it doesn't overwrite the instance variable.

**Comment by tweatherford** [comment]
:+1:

**Comment by tslogan** [comment]
:shipit:

File: lib/greeve/base_item.rb (hunk 1 above)
Comment (amclain) [inline lib/greeve/base_item.rb]: Fixed. This has been moved to `#initialize` so that the merge only happens once.

**Comment by amclain** [comment]
Thanks!

**Pull request merged**